numpy/core/src/multiarray/vec_string.c

```diff
@@ -0,0 +1,365 @@
+     * require a broadcast iterator (and broadcast iterators don't work
+     * with 1 argument anyway).
+     */
+    PyArrayIterObject* in_iter = NULL;
+    PyArrayObject* result = NULL;
+    PyArrayIterObject* out_iter = NULL;
+
+    in_iter = (PyArrayIterObject*)PyArray_IterNew((PyObject*)char_array);
+    if (in_iter == NULL) {
+        Py_DECREF(type);
+        goto err;
+    }
+
+    result = (PyArrayObject*)PyArray_SimpleNewFromDescr(
+            PyArray_NDIM(char_array), PyArray_DIMS(char_array), type);
+    if (result == NULL) {
+        goto err;
+    }
+
+    out_iter = (PyArrayIterObject*)PyArray_IterNew((PyObject*)result);
+    if (out_iter == NULL) {
+        goto err;
+    }
+
+    while (PyArray_ITER_NOTDONE(in_iter)) {
+        if (fast) {
+            if (!((_vec_string_fast_op) method)(&in_iter, out_iter)) {
+                goto err;
+            }
+        } else {
+            PyObject* item_result;
+            PyObject* item = PyArray_ToScalar(in_iter->dataptr, in_iter->ao);
+            if (item == NULL) {
+                goto err;
+            }
+
+            item_result = PyObject_CallFunctionObjArgs((PyObject *) method, item, NULL);
+            Py_DECREF(item);
+
+            if (item_result == NULL) {
+                goto err;
+            }
+
+            if (PyArray_SETITEM(result, PyArray_ITER_DATA(out_iter), item_result)) {
+                Py_DECREF(item_result);
+                PyErr_SetString( PyExc_TypeError,
+                    "result array type does not match underlying function");
+                goto err;
+            }
+            Py_DECREF(item_result);
+        }
+
+        PyArray_ITER_NEXT(in_iter);
+        PyArray_ITER_NEXT(out_iter);
+    }
+
+    Py_DECREF(in_iter);
+    Py_DECREF(out_iter);
+
+    return (PyObject*)result;
+
+ err:
+    Py_XDECREF(in_iter);
+    Py_XDECREF(out_iter);
+    Py_XDECREF(result);
+
+    return 0;
+}
+
+NPY_NO_EXPORT PyObject *
+_vec_string(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *NPY_UNUSED(kwds))
+{
+    PyArrayObject* char_array = NULL;
+    PyArray_Descr *type;
+    PyObject* method_name;
+    PyObject* args_seq = NULL;
+
+    int user_defined_dtype = 0;
+    int fast_method_found = 0;
+    int fast_method_with_args = 0;
+    _vec_string_fast_op fast_method = NULL;
+    PyObject* method = NULL;
+    PyObject* result = NULL;
+
+    if (!PyArg_ParseTuple(args, "O&O&O|O",
+                PyArray_Converter, &char_array,
+                PyArray_DescrConverter, &type,
+                &method_name, &args_seq)) {
+        goto err;
+    }
+
+    user_defined_dtype = _is_user_defined_string_array(char_array);
+    fast_method_found = get_fast_op(char_array, method_name, &fast_method, &fast_method_with_args);
+    if (fast_method_found && !user_defined_dtype) {
+        if (fast_method_with_args) {
+            result = _vec_string_with_args(char_array, type, fast_method, args_seq, /*fast*/ 1);
```

Comment:
Maybe use variables named like `int fast = 1; int slow = 0` and then use them instead of the comment.